backup/backup-cli

```diff
@@ -1,54 +1,61 @@
 #!/usr/bin/env python3
-from backup import FileBackend, get_backend, Change
+from backends import get_backend
+from backend import Change
+from server import SocketServer
+
 import os
 import click
 import json
+import logging
 import sqlite3
 import sys
 
+root = logging.getLogger()
+root.setLevel(logging.INFO)
+
+handler = logging.StreamHandler(sys.stdout)
+handler.setLevel(logging.DEBUG)
+formatter = logging.Formatter('%(message)s')
+handler.setFormatter(formatter)
+root.addHandler(handler)
 
 @click.command()
-@click.argument("lightning-dir", type=click.Path(exists=True))
 @click.argument("backend-url")
+@click.option('--lightning-dir', type=click.Path(exists=True), default=None, help='Use an existing lightning directory (default: initialize an empty backup).')
```

Comment:
Making this an option, but then requiring it to be set seems like a bad idea. We could make it default to the default lightning-dir, in which case we could really make this an option.

Reply:
No, it's not required. The whole idea is that you can leave out the option to make an empty backup (without an initial snapshot). This is useful for initializing the file backend on the server side.

Reply:
Ah I see now. That's clever indeed :+1: I must've missed that. So the first change that the backup plugin must stream is a snapshot then, correct? Is that signalled implicitly via the data version being 0 or do we have to remember this?

Reply:
It doesn't make an automatic snapshot. I suppose this could be done, but it's not :slightly_smiling_face:  It's a manual step right now as documented here, while initializing the `socket:` backend: https://github.com/lightningd/plugins/blob/7c8e6dcf0dde42a836dd49dd4f49d4c9b1d9e416/backup/remote.md#usage

Reply:
Ah I see, so this effectively breaks if we chose the `file://` backend, create an empty backup, and the plugin will kill `lightningd` at startup. Not really ergonomic, but I think we can solve it by special-casing `data_version = 0` and taking a snapshot followed by any eventual incremental change we might have.

Reply:
I'm not sure it's that bad. I mean, to get a `backup.lock` installed in the lightning directory you'd normally want to initialize your `socket:` backend (on the client side)? It's no different from initializing a `file://` backend for use with lightingd.